django-s3-file-field-client/src/client.ts

```diff
@@ -114,6 +122,7 @@ export default class S3FFClient {
    * @param multipartInfo Signed information returned from /upload-complete/.
    */
   protected async finalize(multipartInfo: MultipartInfo): Promise<string> {
+    this.onProgress({ state: 'finalizing' });
```

Comment:
This should start at the beginning of `completeUpload`, since that's part of the asynchronous "overhead" work.

I think you could move this invocation to `uploadFile`, since it doesn't really need to be inside the `protected` helper.